js/block-editor.js

```diff
@@ -153,8 +98,74 @@ jQuery(
 								}
 							}
 						);
+						blockEditorSavePostAndReloadPage();
 					}
 				);
+
+				/**
+				 * Reload the block editor page for empty posts.
+				 *
+				 * @param {string} lang The target language code.
+				 */
+				function reloadPageForEmptyPost( lang ) {
+					// Change the new_lang parameter with the new language value for reloading the page
+					// WPCS location.search is never written in the page, just used to relaoad page ( See line 94 ) with the right value of new_lang
+					// new_lang input is controlled server side in PHP. The value come from the dropdown list of language returned and escaped server side.
+					// Notice that window.location changing triggers automatically page reloading.
+					if ( -1 != location.search.indexOf( 'new_lang' ) ) {
+						// use regexp non capturing group to replace new_lang parameter no matter where it is and capture other parameters which can be behind it
+						window.location.search = window.location.search.replace( /(?:new_lang=[^&]*)(&)?(.*)/, 'new_lang=' + lang + '$1$2' ); // phpcs:ignore WordPressVIPMinimum.JS.Window.location, WordPressVIPMinimum.JS.Window.VarAssignment
+					} else {
+						window.location.search = window.location.search + ( ( -1 != window.location.search.indexOf( '?' ) ) ? '&' : '?' ) + 'new_lang=' + lang; // phpcs:ignore WordPressVIPMinimum.JS.Window.location, WordPressVIPMinimum.JS.Window.VarAssignment
+					}
+				};
+
+				/**
+				 * Triggers block editor post save and reload the block editor page when everything is ok.
+				 */
+				blockEditorSavePostAndReloadPage = function() {
+
+					let unsubscribe = null;
+
+					// Listen if the savePost is completely done by subscribing to its events.
+					const savePostIsDone = new Promise(
+						function( resolve, reject ) {
+							unsubscribe = subscribe(
+								function() {
+									const isSavePostSucceeded = select( 'core/editor' ).didPostSaveRequestSucceed();
+									const isSavePostFailed = select( 'core/editor' ).didPostSaveRequestFail();
+									if ( isSavePostSucceeded || isSavePostFailed ) {
+										if ( isSavePostFailed ) {
+											reject();
+										} else {
+											resolve();
+										}
+									}
+								}
+							);
+						}
+					);
+
+					// Triggers the post save.
+					dispatch( 'core/editor' ).savePost();
```

Comment:
Do we need to save the post again, now that the two actions are done sequentially?